shipyard-controller/handler/debughandler.go

```diff
@@ -0,0 +1,221 @@
+	totalCount := len(projects)
+	if paginationInfo.NextPageKey < int64(totalCount) {
+		payload.Projects = append(payload.Projects, projects[paginationInfo.NextPageKey:paginationInfo.EndIndex]...)
+	}
+
+	payload.TotalCount = float64(totalCount)
+	payload.NextPageKey = paginationInfo.NewNextPageKey
+	c.JSON(http.StatusOK, payload)
+}
+
+// GetAllSequencesForProject godoc
+// @Summary      Get all sequences for specific project
+// @Description  Get all the sequences which are present in a project
+// @Tags         Sequence
+// @Param        project              path      string                    true "The name of the project"
+// @Success      200                  {object}  []models.SequenceState    "ok"
+// @Failure      400                  {object}  models.Error              "Bad Request"
+// @Failure      404                  {object}  models.Error              "not found"
+// @Failure      500                  {object}  models.Error              "Internal error"
+// @Router       /debug/project/{project} [get]
+func (dh *DebugHandler) GetAllSequencesForProject(c *gin.Context) {
+	projectName := c.Param("project")
+	payload, err := dh.DebugManager.GetAllSequencesForProject(projectName)
+
+	if err != nil {
+		if errors.Is(err, ErrProjectNotFound) {
+			SetNotFoundErrorResponse(c, fmt.Sprintf(ProjectNotFoundMsg, projectName))
+			return
+		}
+
+		SetInternalServerErrorResponse(c, fmt.Sprintf(InvalidRequestFormatMsg, err.Error()))
+		return
+	}
+
+	c.JSON(http.StatusOK, payload)
+}
+
+// GetSequenceByID godoc
+// @Summary      Get a sequence with the shkeptncontext
+// @Description  Get a specific sequence of a project which is identified by the shkeptncontext
+// @Tags         Sequence
+// @Param        project              path      string                    true  "The name of the project"
+// @Param        shkeptncontext       path      string                    true  "The shkeptncontext"
+// @Success      200                  {object}  models.SequenceState      "ok"
+// @Failure      400                  {object}  models.Error              "Bad Request"
+// @Failure      404                  {object}  models.Error              "not found"
+// @Failure      500                  {object}  models.Error              "Internal error"
+// @Router       /debug/project/{project}/shkeptncontext/{shkeptncontext} [get]
+func (dh *DebugHandler) GetSequenceByID(c *gin.Context) {
+	shkeptncontext := c.Param("shkeptncontext")
+	projectName := c.Param("project")
+	sequence, err := dh.DebugManager.GetSequenceByID(projectName, shkeptncontext)
+
+	if err != nil {
+		if errors.Is(err, ErrProjectNotFound) {
+			SetNotFoundErrorResponse(c, fmt.Sprintf(ProjectNotFoundMsg, projectName))
+			return
+		}
+
+		if errors.Is(err, ErrSequenceNotFound) {
+			SetNotFoundErrorResponse(c, fmt.Sprintf(UnableFindSequenceMsg, shkeptncontext))
+			return
+		}
+
+		SetInternalServerErrorResponse(c, fmt.Sprintf(InvalidRequestFormatMsg, err.Error()))
```

Comment:
let's use a different error message here, as the InvalidRequestFormatMsg is intended for 400 - Bad Request responses